tests/e2e/tests/simple/simple1_test.go

```diff
@@ -0,0 +1,198 @@
+}
+
+func TestSvc2Svc(t *testing.T) {
+	ns := tc.Kube.Namespace
+	// Get the 2 pods
+	podList, err := getPodList(ns, "app=echosrv")
+	if err != nil {
+		t.Fatalf("kubectl failure to get pods %v", err)
+	}
+	if len(podList) != 2 {
+		t.Fatalf("Unexpected to get %d pods when expecting 2. got %v", len(podList), podList)
+	}
+	// call into the service from each of the pods
+	for _, pod := range podList {
+		glog.Infof("From pod \"%s\"", pod)
+		_, err := util.Shell("kubectl exec -n %s %s -c echosrv -- /usr/local/bin/fortio load -qps 0 -t 10s http://echosrv.%s:8080/echo", ns, pod, ns)
+		if err != nil {
+			t.Fatalf("kubectl failure to run fortio %v", err)
+		}
+	}
+	// Success
+}
+
+func TestAuth(t *testing.T) {
+	ns := tc.Kube.Namespace
+	// Get the 2 pods
+	podList, err := getPodList(ns, "app=fortio-noistio")
+	if err != nil {
+		t.Fatalf("kubectl failure to get pods %v", err)
+	}
+	if len(podList) != 1 {
+		t.Fatalf("Unexpected to get %d pods when expecting 1. got %v", len(podList), podList)
+	}
+	pod := podList[0]
+	glog.Infof("From client, non istio injected pod \"%s\"", pod)
+	res, err := util.Shell("kubectl exec -n %s %s -- /usr/local/bin/fortio load -qps 5 -t 1s http://echosrv.%s:8080/echo", ns, pod, ns)
+	if tc.Kube.AuthEnabled {
+		if err == nil {
+			t.Fatalf("Running with auth on yet able to connect from non istio to istio (insecure): %v", res)
+		} else {
+			glog.Infof("Got expected error with auth on and non istio->istio connection: %v", err)
+		}
+	} else {
+		if err == nil {
+			glog.Infof("Got expected success with auth off and non istio->istio connection: %v", res)
+		} else {
+			t.Fatalf("Unexpected error connect from non istio to istio without auth: %v", err)
+		}
+	}
+}
+
+type fortioTemplate struct {
+	FortioImage string
+}
+
+func getPodList(namespace string, selector string) ([]string, error) {
+	pods, err := util.Shell("kubectl get pods -n %s -l %s -o jsonpath={.items[*].metadata.name}", namespace, selector)
+	if err != nil {
+		return nil, err
+	}
+	return strings.Split(pods, " "), nil
+}
+
+func setTestConfig() error {
+	cc, err := framework.NewCommonConfig("simple_auth_test")
+	if err != nil {
+		return err
+	}
+	tc = new(testConfig)
+	tc.CommonConfig = cc
+	hub := os.Getenv("FORTIO_HUB")
+	tag := os.Getenv("FORTIO_TAG")
+	image := hub + "/fortio:" + tag
+	if hub == "" || tag == "" {
+		image = "ldemailly/fortio:latest" // TODO: change
+	}
+	glog.Infof("Fortio hub %s tag %s -> image %s", hub, tag, image)
+	services := []framework.App{
+		{
+			KubeInject:      true,
+			AppYamlTemplate: util.GetResourcePath(servicesYaml),
+			Template: &fortioTemplate{
+				FortioImage: image,
+			},
+		},
+		{
+			KubeInject:      false,
+			AppYamlTemplate: util.GetResourcePath(nonInjectedYaml),
+			Template: &fortioTemplate{
+				FortioImage: image,
+			},
+		},
+	}
+	for i := range services {
+		tc.Kube.AppManager.AddApp(&services[i])
+	}
+	return nil
+}
+
+func check(err error, msg string) {
```

Comment:
nit: this is only used twice, in main(). it might make sense to just inline the code there, as this doesn't represent a big savings and is arguably a bit of an anti-pattern in golang.

Reply:
I copied this part from mixer_test :-) I think we should move it to the framework if we're going to use it a lot (or inline) ?

Reply:
deleting it